diff --git a/include/vncauth.h b/include/vncauth.h
index 86dc455..b66db2e 100644
--- a/include/vncauth.h
+++ b/include/vncauth.h
@@ -25,6 +25,8 @@
 #define CHALLENGESIZE 16
 
 extern int vncEncryptAndStorePasswd(char *passwd, char *fname);
+extern int vncEncryptAndStorePasswd2(char *passwd, char *passwdViewOnly, char *fname);
 extern char *vncDecryptPasswdFromFile(char *fname);
+extern int vncDecryptPasswdFromFile2(char *fname, char *passwdFullControl, char *passwdViewOnly);
 extern void vncRandomBytes(unsigned char *bytes);
 extern void vncEncryptBytes(unsigned char *bytes, char *passwd);

diff --git a/libvncauth/vncauth.c b/libvncauth/vncauth.c
index a92d322..15a5fbc 100644
--- a/libvncauth/vncauth.c
+++ b/libvncauth/vncauth.c
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <time.h>
 #include <unistd.h>
 #include <vncauth.h>
 #include <d3des.h>

diff --git a/vncrec/misc.c b/vncrec/misc.c
index e71ad54..232457c 100644
--- a/vncrec/misc.c
+++ b/vncrec/misc.c
@@ -25,6 +25,7 @@
 #include <vncviewer.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <sys/wait.h>
 
 static void CleanupSignalHandler(int sig);
 static int CleanupXErrorHandler(Display *dpy, XErrorEvent *error);

diff --git a/vncrec/vncviewer.h b/vncrec/vncviewer.h
index a4d8c82..4b2394e 100644
--- a/vncrec/vncviewer.h
+++ b/vncrec/vncviewer.h
@@ -146,6 +146,7 @@ extern void SetVisualAndCmap();
 
 /* cursor.c */
 
+extern Bool HandleCursorPos(int x, int y);
 extern Bool HandleCursorShape(int xhot, int yhot, int width, int height,
                               CARD32 enc);
 extern void SoftCursorLockArea(int x, int y, int w, int h);
